ntbs-service/Pages/Search/Index.cshtml.cs

```diff
@@ -102,41 +112,118 @@ public async Task<IActionResult> OnGetAsync(int? pageIndex)
+            var notificationss = allPossibleNotifications
+                .OrderByDescending(n => n.NotificationStatus == NotificationStatus.Draft)
+                .ThenByDescending(n => n.NotificationDate ?? n.CreationDate)
+                .ThenByDescending(n => n.NotificationId)
+                .ThenBy(n => n.LTBRID ?? n.ETSID)
+                .Take(PaginationParameters.PageSize);
+            return notificationss;  
+        }
+
         public ContentResult OnGetValidateSearchProperty(string key, string value)
         {
             return ValidationService.ValidateProperty(this, key, value);
         }
 
-        public void SetPaginationDetails()
+        public void SetPaginationDetails(int? nextNtbsOffset, int? nextLegacyOffset, 
+            int? previousNtbsOffset, int? previousLegacyOffset, int? ntbsOffset, int? legacyOffset)
         {
             var queryString = Request.Query;
             var previousPageQueryString = new Dictionary<string, string>();
             foreach (var key in queryString.Keys)
             {
-                previousPageQueryString[key] = queryString[key].ToString();
+                // Copy full query string over apart from any offset values
+                if(!key.Contains("Offset"))
+                {
+                    previousPageQueryString[key] = queryString[key].ToString();
+                }
             }
             var nextPageQueryString = previousPageQueryString;
             if (SearchResults?.HasPreviousPage ?? false)
             {
                 PreviousPageText = "Page " + (SearchResults.PageIndex - 1) + " of " + (SearchResults.TotalPages);
                 previousPageQueryString["pageIndex"] = "" + (SearchResults.PageIndex - 1);
+                if (previousNtbsOffset != null && previousLegacyOffset != null) 
+                {
+                    previousPageQueryString["ntbsOffset"] = "" + previousNtbsOffset;
+                    previousPageQueryString["legacyOffset"] = "" + previousLegacyOffset;
+                }
                 PreviousPageUrl = QueryHelpers.AddQueryString("/Search", previousPageQueryString);
             }
             if (SearchResults?.HasNextPage ?? false)
             {
                 NextPageText = "Page " + (SearchResults.PageIndex + 1) + " of " + (SearchResults.TotalPages);
                 nextPageQueryString["pageIndex"] = "" + (SearchResults.PageIndex + 1);
+                if(nextNtbsOffset != null && nextLegacyOffset != null)
+                {
+                    nextPageQueryString["legacyOffset"] = "" + nextLegacyOffset;
+                    nextPageQueryString["ntbsOffset"] = "" + nextNtbsOffset;
+                }
+                if (ntbsOffset != null && legacyOffset != null) 
+                {
+                    nextPageQueryString["previousNtbsOffset"] = "" + ntbsOffset;
+                    nextPageQueryString["previousLegacyOffset"] = "" + legacyOffset;
+                }

```

Comment:
This method has also grown a little too big to be understood in one gulp. Quite a bit of repetition here, too, that seems prime for commonising?